### PR TITLE
chore: add client without Project

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
There are some non-standard usages in urls.go files, which makes it impossible to parse the right API URL.

1. set the dcs and dms client to without projectID
2. fix test of DCS : the vpc-Datasource need to specify enterprise_project_id

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed
```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run=TestAccDcsAZV1DataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run=TestAccDcsAZV1DataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsAZV1DataSource_basic
=== PAUSE TestAccDcsAZV1DataSource_basic
=== CONT  TestAccDcsAZV1DataSource_basic
--- PASS: TestAccDcsAZV1DataSource_basic (12.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        13.058s
```
```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run=TestAccDcsProductV1DataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run=TestAccDcsProductV1DataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsProductV1DataSource_basic
=== PAUSE TestAccDcsProductV1DataSource_basic
=== CONT  TestAccDcsProductV1DataSource_basic
--- PASS: TestAccDcsProductV1DataSource_basic (13.43s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        13.523s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDcsMaintainWindowDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDcsMaintainWindowDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDcsMaintainWindowDataSource_basic
=== PAUSE TestAccDcsMaintainWindowDataSource_basic
=== CONT  TestAccDcsMaintainWindowDataSource_basic
--- PASS: TestAccDcsMaintainWindowDataSource_basic (7.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       8.022s

```
```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDataSourceDcsFlavorsV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDataSourceDcsFlavorsV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDcsFlavorsV2_basic
=== PAUSE TestAccDataSourceDcsFlavorsV2_basic
=== CONT  TestAccDataSourceDcsFlavorsV2_basic
--- PASS: TestAccDataSourceDcsFlavorsV2_basic (11.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       11.615s

```
```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDcsInstances'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDcsInstances -timeout 360m -parallel 4
=== RUN   TestAccDcsInstances_basic
=== PAUSE TestAccDcsInstances_basic
=== RUN   TestAccDcsInstances_withEpsId
=== PAUSE TestAccDcsInstances_withEpsId
=== RUN   TestAccDcsInstances_whitelists
=== PAUSE TestAccDcsInstances_whitelists
=== RUN   TestAccDcsInstances_tiny
=== PAUSE TestAccDcsInstances_tiny
=== RUN   TestAccDcsInstances_single
=== PAUSE TestAccDcsInstances_single
=== CONT  TestAccDcsInstances_basic
=== CONT  TestAccDcsInstances_tiny
=== CONT  TestAccDcsInstances_whitelists
=== CONT  TestAccDcsInstances_withEpsId
    acceptance.go:347: This environment does not support Enterprise Project ID tests
--- SKIP: TestAccDcsInstances_withEpsId (0.00s)
=== CONT  TestAccDcsInstances_single
--- PASS: TestAccDcsInstances_single (64.35s)
--- PASS: TestAccDcsInstances_tiny (75.34s)
--- PASS: TestAccDcsInstances_whitelists (86.51s)
--- PASS: TestAccDcsInstances_basic (108.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       108.786s

```
```
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDcsInstances_withEpsId'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDcsInstances_withEpsId -timeout 360m -parallel 4
=== RUN   TestAccDcsInstances_withEpsId
=== PAUSE TestAccDcsInstances_withEpsId
=== CONT  TestAccDcsInstances_withEpsId
--- PASS: TestAccDcsInstances_withEpsId (74.06s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       74.146s
```
```

 make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsMaintainWindowDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsMaintainWindowDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsMaintainWindowDataSource_basic
=== PAUSE TestAccDmsMaintainWindowDataSource_basic
=== CONT  TestAccDmsMaintainWindowDataSource_basic
--- PASS: TestAccDmsMaintainWindowDataSource_basic (9.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       9.233s

```
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsProductDataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsProductDataSource -timeout 360m -parallel 4
=== RUN   TestAccDmsProductDataSource_basic
=== PAUSE TestAccDmsProductDataSource_basic
=== RUN   TestAccDmsProductDataSource_kafkaVmSpec
=== PAUSE TestAccDmsProductDataSource_kafkaVmSpec
=== RUN   TestAccDmsProductDataSource_rabbitmqSingle
=== PAUSE TestAccDmsProductDataSource_rabbitmqSingle
=== RUN   TestAccDmsProductDataSource_rabbitmqCluster
=== PAUSE TestAccDmsProductDataSource_rabbitmqCluster
=== CONT  TestAccDmsProductDataSource_basic
=== CONT  TestAccDmsProductDataSource_rabbitmqSingle
=== CONT  TestAccDmsProductDataSource_kafkaVmSpec
=== CONT  TestAccDmsProductDataSource_rabbitmqCluster
--- PASS: TestAccDmsProductDataSource_rabbitmqCluster (10.25s)
--- PASS: TestAccDmsProductDataSource_kafkaVmSpec (10.51s)
--- PASS: TestAccDmsProductDataSource_rabbitmqSingle (10.52s)
--- PASS: TestAccDmsProductDataSource_basic (11.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       11.493s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsKafkaInstances'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsKafkaInstances -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaInstances_basic
=== PAUSE TestAccDmsKafkaInstances_basic
=== RUN   TestAccDmsKafkaInstances_withEpsId
=== PAUSE TestAccDmsKafkaInstances_withEpsId
=== RUN   TestAccDmsKafkaInstances_compatible
=== PAUSE TestAccDmsKafkaInstances_compatible
=== CONT  TestAccDmsKafkaInstances_basic
=== CONT  TestAccDmsKafkaInstances_compatible
    acceptance.go:347: This environment does not support Enterprise Project ID tests
--- SKIP: TestAccDmsKafkaInstances_compatible (0.00s)
=== CONT  TestAccDmsKafkaInstances_withEpsId
    acceptance.go:347: This environment does not support Enterprise Project ID tests
--- SKIP: TestAccDmsKafkaInstances_withEpsId (0.00s)
--- PASS: TestAccDmsKafkaInstances_basic (1124.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1124.290s
```
```

make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsKafkaTopic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsKafkaTopic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaTopic_basic
=== PAUSE TestAccDmsKafkaTopic_basic
=== CONT  TestAccDmsKafkaTopic_basic
--- PASS: TestAccDmsKafkaTopic_basic (589.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       589.341s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsRabbitmqInstances'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsRabbitmqInstances -timeout 360m -parallel 4
=== RUN   TestAccDmsRabbitmqInstances_basic
=== PAUSE TestAccDmsRabbitmqInstances_basic
=== RUN   TestAccDmsRabbitmqInstances_withEpsId
=== PAUSE TestAccDmsRabbitmqInstances_withEpsId
=== RUN   TestAccDmsRabbitmqInstances_compatible
=== PAUSE TestAccDmsRabbitmqInstances_compatible
=== CONT  TestAccDmsRabbitmqInstances_basic
=== CONT  TestAccDmsRabbitmqInstances_compatible
=== CONT  TestAccDmsRabbitmqInstances_withEpsId
    acceptance.go:347: This environment does not support Enterprise Project ID tests
--- SKIP: TestAccDmsRabbitmqInstances_withEpsId (0.00s)
--- PASS: TestAccDmsRabbitmqInstances_compatible (663.08s)
--- PASS: TestAccDmsRabbitmqInstances_basic (1027.43s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       1027.553s
```
```
make testacc TEST='./huaweicloud/services/acceptance/deprecated' TESTARGS='-run=TestAccDmsAZDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/deprecated -v -run=TestAccDmsAZDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsAZDataSource_basic
=== PAUSE TestAccDmsAZDataSource_basic
=== CONT  TestAccDmsAZDataSource_basic
--- PASS: TestAccDmsAZDataSource_basic (9.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/deprecated        9.957s```
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_instance_test.go [hunk 1]
```diff
@@ -289,7 +289,8 @@ func testAccDcsV1Instance_epsId(instanceName string) string {
 data "huaweicloud_availability_zones" "test" {}
 
 data "huaweicloud_vpc" "test" {
-  name = "vpc-default"
+  name                  = "vpc-default"
+  enterprise_project_id = "0"
 }
 
 data "huaweicloud_vpc_subnet" "test" {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dcs/resource_huaweicloud_dcs_instance_test.go:296]: suggest replace the data-sources of subnet, vpc as resources.

**Comment by hmbig2** [comment]
make testacc TEST='./huaweicloud/services/acceptance/dcs' TESTARGS='-run=TestAccDataSourceDcsFlavorsV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs -v -run=TestAccDataSourceDcsFlavorsV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDcsFlavorsV2_basic
=== PAUSE TestAccDataSourceDcsFlavorsV2_basic
=== CONT  TestAccDataSourceDcsFlavorsV2_basic
--- PASS: TestAccDataSourceDcsFlavorsV2_basic (8.27s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       8.346s

**Pull request merged**